tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/types/family/TType.java

```diff
@@ -0,0 +1,3 @@
+package org.tensorflow.types.family;
+
+public interface TType {}
```

Comment:
Why do we need empty interfaces like this? That's always a code smell to me... They probably have *some* functionality in common. BTW, we have default methods too starting with Java SE 8. Can we use those?

Reply:
This is just to make sure that not all Java types can be passed as a generic type of a `Tensor`, like it is enforced here:
```
 public static <T extends TType> Tensor<T> create(Object obj, DataType<T> dtype) {...}
```
It very soft compile-time type safety but better than nothing, I think.

Reply:
In a future version of Java we will be able to seal this interface so it only has the implementations that we specify, which will make it more useful & reliable (as we'll be able to switch on the subclasses with exhaustiveness checking, and users won't be able to do their own implementations that our code can't work with).

Reply:
I see, that sounds useful.

Reply:
Question for you @Craigacp, will the sealed interface work in Java the same way as in Kotlin, meaning that will only interfaces from the same package (or the same file for Kotlin) be allowed to extend from it or will there be a different safety mechanism?

Reply:
This is the current JEP for the sealed interfaces feature - https://openjdk.java.net/jeps/360. It requires them to be in the same package or module, and listed in the `permits` clause of the interface/class. If the subclasses are all in the same java file, then they don't have to be listed in the permits clause (but that's probably not how we'll do it).